### PR TITLE
Make the 'command' & node languages use buildpacks

### LINKED ISSUE #905: Support 'command' invoker via buildpacks

### PR DESCRIPTION
**Pull request by ericbottard**
Expose artifact and handler as BuildTemplate parameters

Fixes #905

### CODE REVIEW

**Comment by ericbottard** [comment]
Rebased on latest master

Blocked by https://github.com/projectriff/riff-buildpack-group/pull/3 (or we could push a manual version of the builder for now)

File: main.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ var (
 			},
 			Namespace: []string{
 				"https://storage.googleapis.com/riff-releases/latest/riff-build.yaml",
-				"https://storage.googleapis.com/riff-releases/riff-cnb-buildtemplate-0.1.0.pre.3.yaml",
+				"https://raw.githubusercontent.com/projectriff/riff-buildpack-group/cf202c43b5695a74185929e393e5c1005e6f1f91/riff-cnb-buildtemplate.yaml",
```
Comment (scothis) [inline main.go]: We should keep release files like this on GCS, I uploaded the file as

```suggestion
				"https://storage.googleapis.com/riff-releases/riff-cnb-buildtemplate-0.1.1.pre.1.yaml",
```

File: pkg/core/function.go [hunk 2]
```diff
@@ -133,6 +133,8 @@ func (c *client) CreateFunction(options CreateFunctionOptions, log io.Writer) (*
 					Name: "riff-cnb",
 					Arguments: []build.ArgumentSpec{
 						{Name: "IMAGE", Value: options.Image},
+						{Name: "FUNCTION_ARTIFACT", Value: options.Artifact},
```
Comment (scothis) [inline pkg/core/function.go]: 👍

**Comment by scothis** [comment]
We'll also need to add artifact and handler support for local builds, but we can add that in a separate PR

**Comment by scothis** [comment]
can you run `make docs` and include the updated output?

File: main.go (hunk 1 above)
Comment (ericbottard) [reply main.go]: Agreed. This is really to unstuck the whole thing for now.
As a matter of fact, the CI for the buildpack group could reference timestamped tagged versions of the docker image (and push that to GCS)

File: distro/main.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ var (
 			},
 			Namespace: []string{
 				"https://storage.googleapis.com/riff-releases/latest/riff-build.yaml",
-				"https://storage.googleapis.com/riff-releases/riff-cnb-buildtemplate-0.1.0.yaml",
+				"https://storage.googleapis.com/projectriff/riff-buildtemplate/riff-cnb-buildtemplate-0.0.1-snapshot-ci-64830c3bbc6503beafdae382ead115806fa100ca.yaml",
```
Comment (glyn) [inline distro/main.go]: What will stop us shipping this snapshot as part of a release?

Comment (glyn) [reply distro/main.go]: Including a snapshot in `stable` could cause problems if it's difficult to rebuild at the correct version (which would make it hard to apply a point fix for a bug without getting a load of other changes that are not necessarily desirable).

Including a snapshot in `latest` does seem counter-intuitive as you point out. I guess it depends on the definition of `latest` - does that just apply to knative components or is it meant to include riff components too?

Comment (ericbottard) [reply distro/main.go]: It's a *pinned* snapshot (as seen in the commit sha that is part of the name).
I'm fine switching it to a "tracking" snapshot for now (and then we can pin / release it when we release `riff`)

Comment (scothis) [reply distro/main.go]: I agree with Eric, it's no different than the pinned snapshot builds we use for knative.

**Pull request merged**